crates/inverted/src/operator.rs

```diff
@@ -0,0 +1,40 @@
+use base::operator::*;
+use base::scalar::F32;
+use quantization::operator::OperatorQuantization;
+use storage::OperatorStorage;
+
+use std::iter::{zip, Empty};
+
+pub trait OperatorInvertedSparse: OperatorQuantization + OperatorStorage {
+    fn to_index_vec(vec: Borrowed<'_, Self>) -> impl Iterator<Item = (u32, F32)>;
+}
+
+impl OperatorInvertedSparse for SVecf32Dot {
+    fn to_index_vec(vec: Borrowed<'_, Self>) -> impl Iterator<Item = (u32, F32)> {
+        zip(vec.indexes().to_vec(), vec.values().to_vec())
```

Comment:
`std::iter::zip(vec.indexes().iter(), vec.values().iter())`

Reply:
```
std::iter::zip(vec.indexes().iter().copied(), vec.values().iter().copied())
```

Reply:
borrowed value does not live long enough

Reply:
```
fn to_index_vec(vec: Borrowed<'_, Self>) -> impl Iterator<Item = (u32, F32)> + '_;
```

Reply:
`vec` is borrowed, returns a value referencing data owned by the current function

Reply:
```
diff --git a/crates/base/src/vector/svecf32.rs b/crates/base/src/vector/svecf32.rs
index be71b5f..6a2934f 100644
--- a/crates/base/src/vector/svecf32.rs
+++ b/crates/base/src/vector/svecf32.rs
@@ -138,12 +138,12 @@ impl<'a> SVecf32Borrowed<'a> {
     }
 
     #[inline(always)]
-    pub fn indexes(&self) -> &[u32] {
+    pub fn indexes(&self) -> &'a [u32] {
         self.indexes
     }
 
     #[inline(always)]
-    pub fn values(&self) -> &[F32] {
+    pub fn values(&self) -> &'a [F32] {
         self.values
     }
 
diff --git a/crates/inverted/src/operator.rs b/crates/inverted/src/operator.rs
index 56af6d7..6dd414a 100644
--- a/crates/inverted/src/operator.rs
+++ b/crates/inverted/src/operator.rs
@@ -6,12 +6,12 @@ use storage::OperatorStorage;
 use std::iter::{zip, Empty};
 
 pub trait OperatorInvertedIndex: OperatorQuantization + OperatorStorage {
-    fn to_index_vec(vec: Borrowed<'_, Self>) -> impl Iterator<Item = (u32, F32)>;
+    fn to_index_vec(vec: Borrowed<'_, Self>) -> impl Iterator<Item = (u32, F32)> + '_;
 }
 
 impl OperatorInvertedIndex for SVecf32Dot {
-    fn to_index_vec(vec: Borrowed<'_, Self>) -> impl Iterator<Item = (u32, F32)> {
-        zip(vec.indexes().to_vec(), vec.values().to_vec())
+    fn to_index_vec(vec: Borrowed<'_, Self>) -> impl Iterator<Item = (u32, F32)> + '_ {
+        zip(vec.indexes().iter().copied(), vec.values().iter().copied())
     }
 }
```

Reply:
Fixed